py/dml/set_test.py

```diff
@@ -0,0 +1,99 @@
+# © 2023 Intel Corporation
+# SPDX-License-Identifier: MPL-2.0
+
+from dml.set import Set
+
+import unittest
+
+class TestSet(unittest.TestCase):
+    def test_contains(self):
+        (o1, o2) = (object(), object())
+        x = Set([0, 2, 'foo', o1])
+        self.assertTrue(0 in x)
+        self.assertFalse(1 in x)
+        self.assertTrue('foo' in x)
+        self.assertFalse('bar' in x)
+        self.assertTrue(o1 in x)
+        self.assertFalse(o2 in x)
+
+    def test_eq(self):
+        x = Set(range(10))
+        self.assertEqual(x, Set(range(10)))
+        self.assertNotEqual(x, Set(range(11)))
+        self.assertEqual(x, set(range(10)))
+        assert set(range(10)) == {e: e for e in range(10)}.keys()
```

Comment:
this plain assert sticks out, why is this like this?

Reply:
The class under test is `Set`, and this line doesn't test `Set` but is rather an executable comment that confirms that ordinary sets work the way that the previous tests expect Set to work (to me it's far from obvious that set and dict.keys() are expected to compare equal)